### PR TITLE
scribusUnstable: 2019-01-16 -> 1.5.5

### PR DESCRIPTION
**Pull request by erictapen**
[changelog](https://bugs.scribus.net/changelog_page.php?version_id=106)

Also changed the pname from `scribus-unstable` to just `scribus`, as this seems to be the more widely used pattern.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/office/scribus/unstable.nix [hunk 1]
```diff
@@ -1,26 +1,24 @@
...
-    sha256 = "16xpsbp6kca78jf48n6zdmyjras38xr11paan839hgy4ik83ncn0";
+  src = fetchurl {
+    url = "mirror://sourceforge/${pname}/${pname}-devel/${pname}-${version}.tar.xz";
+    sha256 = "eQiyGmzoQyafWM7fX495GJMlfmIBzOX73ccNrKL+P3E=";
   };
 
   enableParallelBuilding = true;
 
-  nativeBuildInputs = [ wrapQtAppsHook ];
+  nativeBuildInputs = [ pkgconfig cmake wrapQtAppsHook ];
```
Comment (worldofpeace) [inline pkgs/applications/office/scribus/unstable.nix]:
```suggestion
  nativeBuildInputs = [ pkgconfig cmake  ];
```

File: pkgs/applications/office/scribus/unstable.nix [hunk 2]
```diff
@@ -1,26 +1,24 @@
-{ stdenv, fetchsvn, wrapQtAppsHook, pkgconfig, cmake, qtbase, cairo, pixman,
+{ stdenv, fetchurl, wrapQtAppsHook, pkgconfig, cmake, qtbase, cairo, pixman,
 boost, cups, fontconfig, freetype, hunspell, libjpeg, libtiff, libxml2, lcms2,
 podofo, poppler, poppler_data, python2, harfbuzz, qtimageformats, qttools }:
 
 let
   pythonEnv = python2.withPackages(ps: [ps.tkinter ps.pillow]);
-  revision = "22806";
 in
 stdenv.mkDerivation rec {
```
Comment (worldofpeace) [inline pkgs/applications/office/scribus/unstable.nix]:
```suggestion
mkDerivation rec {
```

File: pkgs/applications/office/scribus/unstable.nix [hunk 3]
```diff
@@ -1,26 +1,24 @@
-{ stdenv, fetchsvn, wrapQtAppsHook, pkgconfig, cmake, qtbase, cairo, pixman,
+{ stdenv, fetchurl, wrapQtAppsHook, pkgconfig, cmake, qtbase, cairo, pixman,
```
Comment (worldofpeace) [inline pkgs/applications/office/scribus/unstable.nix]:
```suggestion
{ stdenv, fetchurl, mkDerivation, pkgconfig, cmake, qtbase, cairo, pixman,
```

**Review by worldofpeace — CHANGES_REQUESTED** [review]
> Also changed the pname from scribus-unstable to just scribus, as this seems to be the more widely used pattern.

That is correct, though note since it was off svn having it as `unstable` with date is directive from the manual.

---

Suggestions are how this application should be packaged according to the qt documentation.

File: pkgs/applications/office/scribus/unstable.nix [hunk 4]
```diff
@@ -1,26 +1,24 @@
...
 boost, cups, fontconfig, freetype, hunspell, libjpeg, libtiff, libxml2, lcms2,
 podofo, poppler, poppler_data, python2, harfbuzz, qtimageformats, qttools }:
 
 let
   pythonEnv = python2.withPackages(ps: [ps.tkinter ps.pillow]);
-  revision = "22806";
 in
 stdenv.mkDerivation rec {
-  name = "scribus-unstable-${version}";
-  version = "2019-01-16";
```
Comment (worldofpeace) [inline pkgs/applications/office/scribus/unstable.nix]: It actually appears this  is a downgrade according to the release date
*  2018-04-28

> https://bugs.scribus.net/changelog_page.php?version_id=106

Comment (erictapen) [reply pkgs/applications/office/scribus/unstable.nix]: That's weird. I got the link to the change log from this recent [release announcement on their mailing list](http://lists.scribus.net/pipermail/scribus/2019-August/055703.html). I guess the date 2018-04-28 means, that the changelog was released on the last release 1.5.4 and got appended until recently.
I also checked the checksum in the announcement mail and they match with the tar ball introduced by this PR.

**Comment by erictapen** [comment]
> Suggestions are how this application should be packaged according to the qt documentation.

Could you give a link to that documentation? I couldn't find anything about your changes in the [Nixpkgs docs](https://nixos.org/nixpkgs/manual/#ssec-qt-applications) (or didn't understand it enough).

That wrapQtAppsHook was introduced by 51d78034a1db78f8ce0ea3ba6fa20be590ed0ca2 btw and I don't really understand what it does anyway, so no hard feelings about your suggestions.

**Comment by erictapen** [comment]
Rebased to master and added your suggestions. Thanks for the feedback!

@GrahamcOfBorg build scribusUnstable

**Comment by worldofpeace** [comment]
> > Suggestions are how this application should be packaged according to the qt documentation.
> 
> Could you give a link to that documentation? I couldn't find anything about your changes in the [Nixpkgs docs](https://nixos.org/nixpkgs/manual/#ssec-qt-applications) (or didn't understand it enough).
> 
> That wrapQtAppsHook was introduced by [51d7803](https://github.com/NixOS/nixpkgs/commit/51d78034a1db78f8ce0ea3ba6fa20be590ed0ca2) btw and I don't really understand what it does anyway, so no hard feelings about your suggestions.

See https://hydra.nixos.org/build/96804884/download/1/nixpkgs/manual.html#sec-language-qt

The part about using `mkDerivation` and not using anything from `qt5.*` set.

**Comment by worldofpeace** [comment]
> That's weird. I got the link to the change log from this recent release announcement on their mailing list. I guess the date 2018-04-28 means, that the changelog was released on the last release 1.5.4 and got appended until recently.
I also checked the checksum in the announcement mail and they match with the tar ball introduced by this PR.

Great, I believe you've done all the appropriate checks then.

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build scribusUnstable

**Review by worldofpeace — APPROVED** [review]
Builds and executes for me on x86_64-linux NixOS :+1:

**Comment by worldofpeace** [comment]
Going to skip the x86_64-linux check since I've tested this.

**Comment by erictapen** [comment]
Thanks!

**Pull request merged**